src/common/HTTPRequest.ts

```diff
@@ -295,12 +382,42 @@ export class HTTPRequest {
    * ```
    *
    * @param overrides - optional overrides to apply to the request.
+   * @param priority - (optional, default 0). Priority 0 will execute immediately.
+   * Priorities > 0 will be resolved using coopeative handling rules.
    */
-  async continue(overrides: ContinueRequestOverrides = {}): Promise<void> {
+  async continue(
+    overrides: ContinueRequestOverrides = {},
+    priority?: number
+  ): Promise<void> {
     // Request interception is not supported for data: urls.
     if (this._url.startsWith('data:')) return;
     assert(this._allowInterception, 'Request Interception is not enabled!');
     assert(!this._interceptionHandled, 'Request is already handled!');
+    if (priority === undefined) {
+      return this._continue(overrides);
+    }
+    this._continueRequestOverrides = overrides;
+    if (priority > this._currentPriority) {
+      this._currentStrategy = 'continue';
+      this._currentPriority = priority;
+      return;
+    }
+    if (priority === this._currentPriority) {
+      if (
+        this._currentStrategy === 'abort' ||
+        this._currentStrategy === 'respond'
+      ) {
+        return;
+      }
+      this._currentPriority = priority;
```

Comment:
In this case, `priority` is already equal to `this._currentPriority` because of the `===` comparison earlier.